File: interview-preparation-kit/archive/string_manipulation/alternating_characters.py
#!/bin/python3

import math
import os
import random
import re
import sys

def alternatingCharacters(string):
    # print(len(list(filter(lambda bool: bool==True, [string[index]==string[index+1] for index in range(len(string)-1)]))))
    deletions=0
    for index in range(len(string)-1):
        if string[index]==string[index+1]:
            deletions+=1
    return deletions

if __name__ == '__main__':
    n = int(input())
    for _ in range(n):
        string = input()
        result = alternatingCharacters(string)
        print(str(result)+'\n')

File: interview-preparation-kit/archive/string_manipulation/alternating_characters.js
'use strict';

const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({input: process.stdin, output: process.stdout});

let inputString = [];

rl.prompt();

rl.on('line', (inputLine) => {
    inputString.push(inputLine);
})

rl.on('close', () => {
    main(inputString).forEach((element) => { console.log(element); });
    process.exit(0);
})

function alternatingCharacters(string) {
    // console.log(string.split('').reduce((acc, cur, index, arr) => { console.log(acc, cur, index, arr[index+1], cur == arr[index+1], cur == arr[index+1] ? 1 : 0); return acc + (cur == arr[index+1] ? 1 : 0) }, 0))
    const deletions = string.split('').reduce((acc, cur, index, arr=string) => { return acc + (cur == arr[index+1] ? 1 : 0) }, 0)
    return deletions;
}

function main(inputString) {
    const n_queries = parseInt(inputString[0].split(' '), 10);

    const results = Array(n_queries)
    for (let i=1; i<=n_queries; i++) {
        const result = alternatingCharacters(inputString[i]);
        results.push(result);
    }
    return results
}
